pkg/handler/enqueue_owner.go

```diff
@@ -0,0 +1,165 @@
+func (e *EnqueueOwner) Update(q workqueue.RateLimitingInterface, evt event.UpdateEvent) {
+	for _, req := range e.getOwnerReconcileRequest(evt.MetaOld) {
+		q.AddRateLimited(req)
+	}
+	for _, req := range e.getOwnerReconcileRequest(evt.MetaNew) {
+		q.AddRateLimited(req)
+	}
+}
+
+// Delete implements EventHandler
+func (e *EnqueueOwner) Delete(q workqueue.RateLimitingInterface, evt event.DeleteEvent) {
+	for _, req := range e.getOwnerReconcileRequest(evt.Meta) {
+		q.AddRateLimited(req)
+	}
+}
+
+// Generic implements EventHandler
+func (e *EnqueueOwner) Generic(q workqueue.RateLimitingInterface, evt event.GenericEvent) {
+	for _, req := range e.getOwnerReconcileRequest(evt.Meta) {
+		q.AddRateLimited(req)
+	}
+}
+
+// parseOwnerTypeGroupKind parses the OwnerType into a Group and Kind and caches the result.  Returns false
+// if the OwnerType could not be parsed using the scheme.
+func (e *EnqueueOwner) parseOwnerTypeGroupKind(scheme *runtime.Scheme) error {
+	// Get the kinds of the type
+	kinds, _, err := scheme.ObjectKinds(e.OwnerType)
+	if err != nil {
+		log.Error(err, "Could not get ObjectKinds for OwnerType", "OwnerType", e.OwnerType)
+		return err
+	}
+	// Expect only 1 kind.  If there is more than one kind this is probably an edge case such as ListOptions.
+	if len(kinds) != 1 {
+		err := fmt.Errorf("Expected exactly 1 kind for OwnerType")
+		log.Error(err, "", "OwnerType", e.OwnerType, "Kinds", kinds)
+		return err
+
+	}
+	// Cache the Group and Kind for the OwnerType
+	e.groupKind = schema.GroupKind{Group: kinds[0].Group, Kind: kinds[0].Kind}
+	return nil
+}
+
+// getOwnerReconcileRequest looks at object and returns a slice of reconcile.Request to reconcile
+// owners of object that match e.OwnerType.
+func (e *EnqueueOwner) getOwnerReconcileRequest(object metav1.Object) []reconcile.Request {
+	// Iterate through the OwnerReferences looking for a match on Group and Kind against what was requested
+	// by the user
+	var result []reconcile.Request
+	for _, ref := range e.getOwnersReferences(object) {
+		// Parse the Group out of the OwnerReference to compare it to what was parsed out of the requested OwnerType
+		refGV, err := schema.ParseGroupVersion(ref.APIVersion)
+		if err != nil {
+			log.Error(err, "Could not parse OwnerReference GroupVersion",
+				"OwnerReference", ref.APIVersion)
+			return nil
+		}
+
+		// Compare the OwnerReference Group and Kind against the OwnerType Group and Kind specified by the user.
+		// If the two match, create a Request for the objected referred to by
+		// the OwnerReference.  Use the Name from the OwnerReference and the Namespace from the
+		// object in the event.
+		if ref.Kind == e.groupKind.Kind && refGV.Group == e.groupKind.Group {
+			// Match found - add a Request for the object referred to in the OwnerReference
+			result = append(result, reconcile.Request{NamespacedName: types.NamespacedName{
+				Namespace: object.GetNamespace(),
+				Name:      ref.Name,
+			}})
+		}
```

Comment:
Its good that we are not performing lookup against the cluster to match the UUIDs for the owner because that would have made handling the cases of owner itself changing difficult.